.build/config/cassandra_ci.yaml

```diff
@@ -0,0 +1,342 @@
+      - long-test
+      - cqlsh-test
+
+  # The post-commit pipeline is a larger set of tests that will be matrixed
+  # across other configuration axes (see parameters below) where we expect
+  # different implementations or configurations to only fail rarely if at all.
+  # Failures in tests here will be made visible on JIRA tickets shortly after
+  # test run on reference CI and committers are expected to prioritize
+  # rectifying any failures introduced by their work.
+  - name: post-commit
+    jobs:
+      - unit
+      - cdc
+      - compression
+      - test-oa
+      - test-system-keyspace-directory
+      - test-tries
+      - jvm-dtest
+      - jvm-dtest-upgrade
+      - dtest
+      - dtest-novnode
+      - dtest-offheap
+      - dtest-large
+      - dtest-large-novnode
+      - dtest-upgrade
+      - dtest-upgrade-large
+      - long-test
+      - cqlsh-test
+
+  # These are longer-term, much more rarely changing pieces of infrastructure or
+  # testing. We expect these to fail even more rarely than post-commit.
+  - name: nightly
+    jobs:
+      - stress-test
+      - fqltool-test
+      - test-burn
+
+#-----------------------------------------------------------------------------
+# JOBS
+#
+# Parameters:
+#   job: the name
+#   runtime_env: which runtime_env above to base self off
+#   description: text description of the job
+#   cmd: command to run in shell to execute tests
+#   testlist: command to run in shell to generate full list of tests to run. Expected to output filename value to TESTLIST= env var.
+#   test_timeout: timeout for the test in ms
+#   multiplex: Number of times to multiplex a test of the given suite on addition or change
+#
+# Required env vars:
+#   DIST_DIR=<C*/build dir>
+#   CASSANDRA_DIR=<C* dir>
+#   TESTLIST=<populated by testlist: tag entry on a given job>
+#   JOB=<populated by job: tag entry on a given job>
+#
+# Jobs can replace any base params defined in their resource class, and map
+# types will append new or overwriting keys.
+#
+#-----------------------------------------------------------------------------
+# RUNTIME ENVIRONMENTS
+# i.e. common resource classes
+#-----------------------------------------------------------------------------
+runtime_env:
+  - type: jvm
+    resources: {cpu=4, memory=4g, storage=25g}
+    env:
+      ANT_TEST_OPTS: "-Dno-build-test=true"
+    testlist: find test/unit -name '*Test.java' | sed "s;^test/unit/;;g" | sort
+    test_timeout: 480000
+    multiplex: 500
+
+  - type: jvm-dtest
+    resources: {cpu=4, memory=16g, storage=25g}
+    env:
+      ANT_TEST_OPTS: "-Dno-build-test=true"
+    test_timeout: 900000
+    multiplex: 500
+
+  - type: python
+    resources: {cpu=4, memory=6g, storage=50g}
+    testlist: echo "WARNING! You have to define a testlist entry for this job." && exit 1
+    test_timeout: 900000
+    multiplex: 500
+    cmd: pytest ${PYTEST_OPTS} --cassandra-dir=${CASSANDRA_DIR} --keep-failed-test-dir ${DTEST_ARGS} ${TESTLIST}
+    env:
```

Comment:
are there assumptions about ordering?  can I do ```env:
  FOO: "bar"
  BAR: "${FOO}"``` and have this setup `BAR=bar`?  

Or is the order of applying "undefined"?  I ask as `DTEST_ARGS` has a dependency on `NUM_TOKENS`, which makes this something we need to be explicit about defining

Reply:
From an ordering perspective, ISTM we can document / enforce / expect a "everything else needs to be resolved and parsed before you parse `cmd` and `env`.", and only allow those 2 to interpolate env variables.

There's some hand-waving around "some of these items become environment variables and other env var are declared explicitly in the env: section" I don't currently love. I'm not familiar enough w/yaml to know if there's an idiomatic solution to that or if this is something we should document and punt on.